### DISCUSSION TITLE
Tanix TX3 emmc install error

### DISCUSSION THREAD

**Discussion by EastWoodYang** [devmfc/debian-on-amlogic#38] [Q&A] [answered]
Hi **Devmfc**,
Thanks you for your **great work**.
I tried to install your debian in emmc on my Tanix TX3, but show error: 

![image](https://github.com/devmfc/debian-on-amlogic/assets/9335046/4b3ae46f-9e81-4549-b0f7-615ea8d13a50)

**Comment by EastWoodYang** [comment]
Also, Ethernet is not working,  my Tanix TX3 is 100m, not Gigabit, so I use dtb from https://github.com/ophub/amlogic-s9xxx-armbian/blob/main/build-armbian/armbian-files/platform-files/amlogic/bootfs/dtb/amlogic/meson-sm1-tx3-bz.dtb, and it work.

**Comment by EastWoodYang** [comment]
https://github.com/devmfc/debian-on-amlogic/discussions/28#discussioncomment-8500895

I will try ‘source /etc/boot-run-once/03-set_fw_env_config‘ tonight, hope it work

**Comment by EastWoodYang** [comment]
Run `source /etc/boot-run-once/03-set_fw_env_config` is not working for my box. and `fw_env.config` is empty.

![c47171b9d82da69af526f49c5db4ab7](https://github.com/devmfc/debian-on-amlogic/assets/9335046/e8c3b6b7-b5ae-4e5d-ba93-c1339c9e714e)

**Reply by devmfc** [reply]
It can't find the environment storage for your bootloader. Did you flash some other image to your box before? Can you post the output of `~/aml-part.sh`

**Reply by EastWoodYang** [reply]
Yes, your are right ! I flash [other image](https://github.com/ophub/amlogic-s9xxx-armbian) to my box before. and output of `~/aml-part.sh`: 

![b55b87e08a1fef386e41c39f13ae64b](https://github.com/devmfc/debian-on-amlogic/assets/9335046/f51c1b02-06d3-440a-8b98-a8e27849335f)

**Reply by devmfc** [reply, marked as answer]
My images assume the presence of the stock bootloader. So you have two things you could try:
1. revert to stock android and try again
2. create an fw_env.config for the location of the env partition. But I don't know what bootloader your box is currently using, so the risk of bricking you box is much higher with this option. And I don't have the time to help you unbrick it

**Reply by EastWoodYang** [reply]
I had reset my box to Android, and already install to emmc. thank you so mush!

And could you add Tanix TX3 100m support in the next release ?

**Reply by devmfc** [reply]
Could you test the attached DTB?

[meson-sm1-tanixtx3-100M.zip](https://github.com/devmfc/debian-on-amlogic/files/14594629/meson-sm1-tanixtx3-100M.zip)

**Reply by EastWoodYang** [reply]
after replace 100M dtd, my box can not boot from emmc, but can boot from usb. and now I install to emmc again.

![e70237b4e6df7ca138e2c956120fa86](https://github.com/devmfc/debian-on-amlogic/assets/9335046/4df8d018-395d-4c72-be20-a34ec8a70c18)

**Reply by EastWoodYang** [reply]
Hi devmfc, Where is meson-sm1-tanixtx3-100M dtb source ? I want to change some config.

**Reply by devmfc** [reply]
The dtb you used before has eMMC on /dev/mmcblk2, mainline images have eMMC on /dev/mmcblk1, so if you switch between these two, you'll have to change the config where rootfs is. This is why your box didn't boot any more after changing the dtb.

It is better to try any new dtb, kernel etc on a USB flash drive first.

For the source of the dtb: `dtc meson-sm1-tanixtx3-100M.dtb > meson-sm1-tanixtx3-100M.dts` and the dts will contain the source.
